mcfix_account/models/account_payment.py

```diff
@@ -64,6 +64,15 @@ def _check_company_id(self):
 class AccountAbstractPayment(models.AbstractModel):
     _inherit = 'account.abstract.payment'
 
+    @api.multi
+    def _get_default_company(self):
+        if self.env.context.get('default_invoice_ids'):
+            inv = self.env['account.invoice'].browse(self.env.context['default_invoice_ids'][0][1])
+            return inv.company_id.id
+        return self.env.user.company_id.id
+
+    create_company_id = fields.Many2one('res.company', string='Company', default=_get_default_company)
```

Comment:
Line 70 and 74 are >80 lines and so the linter complains and the PR is red

Reply:
We're not actively maintaing the v10 branch. I believe that by now it must be quite obsolete.

v11 is up to date. I suggest that you consider reviewing that branch first.

Reply:
Unfortunately I have a customer on 10.0 who wants to use this :-)

I have realized 10.0 is behind, so am now testing 11.0 and considering the possibility of backporting some functionalities.